src/core/padding.rs

```diff
@@ -11,7 +11,11 @@ where
 {
     /// Taking in the image data and the margin to apply to rows and columns
     /// returns a padded image
-    fn pad(&self, image: ArrayView3<T>, padding: (usize, usize)) -> Array3<T>;
+    fn pad(
+        &self,
+        image: ArrayView<T, Ix3>,
```

Comment:
I think this is too restrictive - wouldn't it be more convenient to take `&ArrayBase<U, Ix3>` where `U: Data<Elem = T>`? Is there any specific reason you are using a view?

Reply:
So I tried to do it but the generics got a bit hairy especially with the convolution extension trait. Just tried to do it again and got `overflow evaluating the requirement `<ndarray::ArrayBase<U, ndarray::dimension::dim::Dim<[usize; 3]>> as processing::conv::ConvolutionExt<S>>::Data`. It's something I'd like to do but it's a bit of a pain (or the easy way to do it isn't yet evident to me)